circuits/cpp/src/aztec3/circuits/kernel/private/native_private_kernel_circuit_ordering.cpp

```diff
@@ -46,16 +43,14 @@ void match_reads_to_commitments(DummyCircuitBuilder& builder,
     for (size_t rr_idx = 0; rr_idx < MAX_READ_REQUESTS_PER_TX; rr_idx++) {
         const auto& read_request = read_requests[rr_idx];
         const auto& hint_to_commitment = hint_to_commitments[rr_idx];
+        const auto hint_pos = static_cast<size_t>(uint64_t(hint_to_commitment));
 
         if (read_request != 0) {
             size_t match_pos = MAX_NEW_COMMITMENTS_PER_TX;
-            // TODO(https://github.com/AztecProtocol/aztec-packages/issues/892): inefficient
-            // O(n^2) inner loop will be optimized via matching hints
-            for (size_t c_idx = 0; c_idx < MAX_NEW_COMMITMENTS_PER_TX; c_idx++) {
-                match_pos = (read_request == new_commitments[c_idx]) ? c_idx : match_pos;
+            if (hint_pos < MAX_NEW_COMMITMENTS_PER_TX) {
+                match_pos = read_request == new_commitments[hint_pos] ? hint_pos : match_pos;
             }
 
-            // Transient reads MUST match a pending commitment
```

Comment:
This comment was removed?

Reply:
Well now that we use the hint, this comment felt not necessary. If you insist I can ressucitate it.

Reply:
Fair enough!